Using C.NAME for voidptr should require explicit casting

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relvant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** V 0.1.29 8a68714.e06756e
**OS:** linux, "Arch Linux"

**What did you do?**
```v
module main

import dl

#flag -D_GNU_SOURCE

type RealOpen = fn(charptr, int, int) int

[export: open]
fn open(filename charptr, oflag int, mode int) int {
	real_open := RealOpen((dl.sym(C.RTLD_NEXT, "open")))
	return real_open(filename, oflag, mode)
}
```

**What did you expect to see?**
Checker error, the C.RTLD_NEXT needs a voidptr cast.
 
**What did you see instead?**
```
==================
/tmp/v/hook.2822383656832685981.tmp.so.c: In function ‘main__open’:
/tmp/v/hook.2822383656832685981.tmp.so.c:8087:80: error: lvalue required as unary ‘&’ operand
 8087 |  int (*real_open) (charptr , int , int ) = ((main__RealOpen)((dl__sym((voidptr)&/*qq*/RTLD_NEXT, tos_lit("open")))));
      |                                                                                ^
...
==================
```

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -2719,6 +2719,8 @@ fn (mut g Gen) ref_or_deref_arg(arg ast.CallArg, expected_type ast.Type, lang as
 					if expected_type.has_flag(.option) {
 						g.expr_with_opt(arg.expr, arg_typ, expected_type)
 						return
+					} else if arg.expr is ast.Ident && arg.expr.language == .c {
+						g.write('(voidptr)')
 					} else {
 						g.write('(voidptr)&/*qq*/')
 					}

--- a/vlib/v/gen/c/testdata/c_ident_for_ptr_arg.c.must_have
+++ b/vlib/v/gen/c/testdata/c_ident_for_ptr_arg.c.must_have
@@ -1,0 +1,1 @@
+int (*real_open) (charptr , int , int ) = ((main__RealOpen)((dl__sym((voidptr)RTLD_NEXT, _SLIT("open")))));

--- a/vlib/v/gen/c/testdata/c_ident_for_ptr_arg.vv
+++ b/vlib/v/gen/c/testdata/c_ident_for_ptr_arg.vv
@@ -1,0 +1,13 @@
+module main
+
+import dl
+
+#flag -D_GNU_SOURCE
+
+type RealOpen = fn(charptr, int, int) int
+
+@[export: open]
+fn open(filename charptr, oflag int, mode int) int {
+	real_open := RealOpen((dl.sym(C.RTLD_NEXT, "open")))
+	return real_open(filename, oflag, mode)
+}

